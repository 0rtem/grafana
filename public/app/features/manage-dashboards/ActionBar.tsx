import React, { PureComponent } from 'react';
import { hot } from 'react-hot-loader';
import { connect } from 'react-redux';
import { updateSearchQuery } from './state/actions';
import { getCanSave, getFolderId, getHasEditPermissionInFolders, getIsEditor, getSearchQuery } from './state/selectors';

export interface Props {
  searchQuery: string;
  hasEditPermissionInFolders: boolean;
  canSave: boolean;
  isEditor: boolean;
  folderId: number;
  updateSearchQuery: typeof updateSearchQuery;
}

export class ActionBar extends PureComponent<Props> {
  onSearchQueryChange = event => {
    this.props.updateSearchQuery(event.target.value);
  };

  createDashboardUrl = () => {
    const { folderId } = this.props;

    return `dashboard/new${folderId ? '?folderId=' + folderId : ''} `;
  };

  importDashboardUrl = () => {
    const { folderId } = this.props;

    return `dashboard/import${folderId ? '?folderId=' + folderId : ''}`;
  };

  render() {
    const { searchQuery, hasEditPermissionInFolders, canSave, isEditor, folderId } = this.props;

    return (
      <div className="page-action-bar page-action-bar--narrow">
        <label className="gf-form gf-form--grow gf-form--has-input-icon">
          <input
            type="text"
            className="gf-form-input max-width-30"
            placeholder="Find Dashboard by name"
            tabIndex={1}
            give-focus="true"
            value={searchQuery}
            spellCheck={false}
            autoComplete="false"
            autoCorrect="false"
            onChange={this.onSearchQueryChange}
          />
          <i className="gf-form-input-icon fa fa-search" />
        </label>
        <div className="page-action-bar__spacer" />
        {(hasEditPermissionInFolders || canSave) && (
          <a className="btn btn-success" href={this.createDashboardUrl()}>
            <i className="fa fa-plus" /> Dashboard
          </a>
        )}
        {folderId !== undefined &&
          isEditor && (
            <a className="btn btn-success" href="dashboards/folder/new">
              <i className="fa fa-plus" /> Folder
            </a>
          )}
        {hasEditPermissionInFolders || canSave}
        <a className="btn btn-success" href={this.importDashboardUrl()}>
          <i className="fa fa-plus" /> Import
        </a>
      </div>
    );
  }
}

function mapStateToProps(state) {
  return {
    searchQuery: getSearchQuery(state.manageDashboards),
    hasEditPermissionInFolders: getHasEditPermissionInFolders(state.manageDashboards),
    canSave: getCanSave(state.manageDashboards),
    isEditor: getIsEditor(state.manageDashboards),
    folderId: getFolderId(state.manageDashboards),
  };
}

const mapDispatchToProps = {
  updateSearchQuery,
};

export default hot(module)(connect(mapStateToProps, mapDispatchToProps)(ActionBar));
