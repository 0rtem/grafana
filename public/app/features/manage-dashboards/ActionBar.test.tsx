import React from 'react';
import { shallow } from 'enzyme';
import { ActionBar, Props } from './ActionBar';

const setup = (propOverrides?: object) => {
  const props: Props = {
    searchQuery: '',
    hasEditPermissionInFolders: false,
    canSave: false,
    isEditor: false,
    folderId: 0,
    updateSearchQuery: jest.fn(),
  };

  Object.assign(props, propOverrides);

  const wrapper = shallow(<ActionBar {...props} />);
  const instance = wrapper.instance();

  return {
    wrapper,
    instance,
  };
};

describe('Render', () => {
  it('should render component', () => {
    const { wrapper } = setup();

    expect(wrapper).toMatchSnapshot();
  });

  it('should render add dashboard if user has permissions', () => {
    const { wrapper } = setup({
      hasEditPermissionInFolders: true,
    });

    expect(wrapper).toMatchSnapshot();
  });

  it('should render import dashboard if user has permissions', () => {
    const { wrapper } = setup({
      isEditor: true,
    });

    expect(wrapper).toMatchSnapshot();
  });
});
