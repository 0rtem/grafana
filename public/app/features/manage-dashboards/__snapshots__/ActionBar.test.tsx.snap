// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render should render add dashboard if user has permissions 1`] = `
<div
  className="page-action-bar page-action-bar--narrow"
>
  <label
    className="gf-form gf-form--grow gf-form--has-input-icon"
  >
    <input
      autoComplete="false"
      autoCorrect="false"
      className="gf-form-input max-width-30"
      give-focus="true"
      onChange={[Function]}
      placeholder="Find Dashboard by name"
      spellCheck={false}
      tabIndex={1}
      type="text"
      value=""
    />
    <i
      className="gf-form-input-icon fa fa-search"
    />
  </label>
  <div
    className="page-action-bar__spacer"
  />
  <a
    className="btn btn-success"
    href="dashboard/new "
  >
    <i
      className="fa fa-plus"
    />
     Dashboard
  </a>
  <Component />
  <a
    className="btn btn-success"
    href="dashboard/import"
  >
    <i
      className="fa fa-plus"
    />
     Import
  </a>
</div>
`;

exports[`Render should render component 1`] = `
<div
  className="page-action-bar page-action-bar--narrow"
>
  <label
    className="gf-form gf-form--grow gf-form--has-input-icon"
  >
    <input
      autoComplete="false"
      autoCorrect="false"
      className="gf-form-input max-width-30"
      give-focus="true"
      onChange={[Function]}
      placeholder="Find Dashboard by name"
      spellCheck={false}
      tabIndex={1}
      type="text"
      value=""
    />
    <i
      className="gf-form-input-icon fa fa-search"
    />
  </label>
  <div
    className="page-action-bar__spacer"
  />
  <a
    className="btn btn-success"
    href="dashboard/import"
  >
    <i
      className="fa fa-plus"
    />
     Import
  </a>
</div>
`;

exports[`Render should render import dashboard if user has permissions 1`] = `
<div
  className="page-action-bar page-action-bar--narrow"
>
  <label
    className="gf-form gf-form--grow gf-form--has-input-icon"
  >
    <input
      autoComplete="false"
      autoCorrect="false"
      className="gf-form-input max-width-30"
      give-focus="true"
      onChange={[Function]}
      placeholder="Find Dashboard by name"
      spellCheck={false}
      tabIndex={1}
      type="text"
      value=""
    />
    <i
      className="gf-form-input-icon fa fa-search"
    />
  </label>
  <div
    className="page-action-bar__spacer"
  />
  <a
    className="btn btn-success"
    href="dashboards/folder/new"
  >
    <i
      className="fa fa-plus"
    />
     Folder
  </a>
  <a
    className="btn btn-success"
    href="dashboard/import"
  >
    <i
      className="fa fa-plus"
    />
     Import
  </a>
</div>
`;
